sdk/nodejs/runtime/rpc.ts

```diff
@@ -340,7 +361,34 @@ export async function serializeProperty(ctx: string, prop: Input<any>, dependent
         // which will wrap undefined, if it were to be resolved (since `Output` has no member named .isSecret).
         // so we must compare to the literal true instead of just doing await prop.isSecret.
         const isSecret = await prop.isSecret === true;
-        const value = await serializeProperty(`${ctx}.id`, prop.promise(), dependentResources);
+        const value = await serializeProperty(`${ctx}.id`, prop.promise(), dependentResources, {
+            keepOutputValues: false,
+        });
+
+        if (opts?.keepOutputValues && await monitorSupportsOutputValues()) {
+            const dependencies = new Set<string>();
+            for (const resource of propResources) {
+                const urn = await serializeProperty(`${ctx} dependency`, resource.urn, new Set<Resource>(), {
+                    keepOutputValues: false,
+                });
+                dependencies.add(urn);
+            }
```

Comment:
I don't think this is right, but just want to confirm.

@pgavlin, we need to do the same dependency expansion here, as we do here:

https://github.com/pulumi/pulumi/blob/c247f6e283fb94cbce5a3e73ebe226e34ed99d31/sdk/nodejs/runtime/resource.ts#L552-L558

Correct?

Otherwise, the property dependencies for a property may contain more dependencies (expanded) than the union of all nested output value dependencies for the property, when those sets should be equal.

Reply:
I'm going to go ahead and add a commit that does the expansion.